Core/HLE/sceKernelInterrupt.cpp

```diff
@@ -742,10 +742,169 @@ static int sysclib_memcmp(u32 dst, u32 src, u32 size) {
 }
 
 static int sysclib_sprintf(u32 dst, u32 fmt) {
-	ERROR_LOG(SCEKERNEL, "Unimpl sysclib_sprintf(dest=%08x, src=%08x)", dst, fmt);
+	ERROR_LOG(SCEKERNEL, "Untested sysclib_sprintf(dst=%08x, fmt=%08x)", dst, fmt);
 	if (Memory::IsValidAddress(dst) && Memory::IsValidAddress(fmt)) {
-		// TODO: Properly use the format string with more parameters.
-		return sprintf((char *)Memory::GetPointerUnchecked(dst), "%s", Memory::GetCharPointerUnchecked(fmt));
+
+
+		u32 psp_stack_pointer = currentMIPS->r[MIPS_REG_SP];
+		ERROR_LOG(SCEKERNEL, "Fmt: %s", Memory::GetCharPointerUnchecked(fmt));
+		ERROR_LOG(SCEKERNEL, "PSP arg reg dump: 0x%08x 0x%08x 0x%08x 0x%08x 0x%08x 0x%08x 0x%08x 0x%08x",
+		currentMIPS->r[MIPS_REG_A0],
+		currentMIPS->r[MIPS_REG_A1],
+		currentMIPS->r[MIPS_REG_A2],
+		currentMIPS->r[MIPS_REG_A3],
+		currentMIPS->r[MIPS_REG_T0],
+		currentMIPS->r[MIPS_REG_T1],
+		currentMIPS->r[MIPS_REG_T2],
+		currentMIPS->r[MIPS_REG_T3]
+		);
+		for(int i = 0;i < 24; i+=8){
+			u32 base = psp_stack_pointer + i * 4;
+			ERROR_LOG(SCEKERNEL, "PSP stack dump: 0x%08x 0x%08x 0x%08x 0x%08x 0x%08x 0x%08x 0x%08x 0x%08x",
+			Memory::Read_U32(base),
+			Memory::Read_U32(base + 1 * 4),
+			Memory::Read_U32(base + 2 * 4),
+			Memory::Read_U32(base + 3 * 4),
+			Memory::Read_U32(base + 4 * 4),
+			Memory::Read_U32(base + 5 * 4),
+			Memory::Read_U32(base + 6 * 4),
+			Memory::Read_U32(base + 7 * 4)
+			);
+		}
+		int state = 0;
+		std::string result = "";
+		std::string cur_fmt = "";
+		int bytes_to_read = 0;
+		int arg_idx = 0;
+		int fmt_len = 0;
+
+		for(const char *c = Memory::GetCharPointerUnchecked(fmt); *c != '\0'; c++){
```

Comment:
Please at least roughly follow PPSSPP's general code style throughout. You're close, but we write `for(` as `for (`, and ) and { are separated by a space (last part of the line).